spec/unit/actions/app_apply_manifest_spec.rb

```diff
@@ -848,35 +849,182 @@ module VCAP::CloudController
+                  expect(app_apply_manifest).to receive(:sleep).with(1).exactly(3).times
+
+                  app_apply_manifest.apply(app.guid, message)
+                end
+
+                it 'polls service bindings with the default sleep value' do
+                  allow(service_cred_binding_create).to receive(:poll).and_return(V3::ServiceBindingCreate::ContinuePolling.call(nil),
+                                                                                  V3::ServiceBindingCreate::ContinuePolling.call(nil),
+                                                                                  V3::ServiceBindingCreate::PollingFinished,
+                                                                                  V3::ServiceBindingCreate::ContinuePolling.call(nil),
+                                                                                  V3::ServiceBindingCreate::PollingFinished)
+
+                  expect(service_cred_binding_create).to receive(:poll).exactly(5).times
+                  expect(app_apply_manifest).to receive(:sleep).with(5).exactly(3).times
+
+                  app_apply_manifest.apply(app.guid, message)
+                end
+
+                it 'verifies exception is thrown if maximum polling duration is exceeded' do
+                  TestConfig.override(max_manifest_service_binding_poll_duration: 15)
+                  allow_any_instance_of(AppApplyManifest).to receive(:sleep) do |action, seconds|
+                    Timecop.travel(seconds.from_now)
+                  end
+                  allow(service_cred_binding_create).to receive(:poll).and_return(V3::ServiceBindingCreate::ContinuePolling.call(20.seconds),
+                                                                                  V3::ServiceBindingCreate::ContinuePolling.call(20.seconds),
+                                                                                  V3::ServiceBindingCreate::PollingFinished)
+                  expect { app_apply_manifest.apply(app.guid, message) }.to raise_error(AppApplyManifest::ServiceBindingError)
+
+                  expect(binding1.last_operation.state).to eq('failed')
+                  expect(binding1.last_operation.description).to eq('Polling exceed the maximum polling duration')
+
+                  orphan_mitigation_job = Delayed::Job.first
+                  expect(orphan_mitigation_job).not_to be_nil
+                  expect(orphan_mitigation_job).to be_a_fully_wrapped_job_of Jobs::Services::DeleteOrphanedBinding
+                end
+
+                it 'has a maximum retry_after' do
+                  allow(service_cred_binding_create).to receive(:poll).and_return(V3::ServiceBindingCreate::ContinuePolling.call(24.hours),
+                                                                                  V3::ServiceBindingCreate::PollingFinished)
+                  expect(app_apply_manifest).to receive(:sleep).with(60)
+
+                  Timecop.scale(600) do
```

Comment:
Is this needed? What does it do?

Reply:
It makes time go faster by a factor of 600.  It's not actually needed for this test, so I'll remove